### ISSUE TITLE
Options docs out of date?

### ISSUE THREAD

**Issue by richburdon** [OptimalBits/bull#529]
A few issues getting started:

1). According to the docs (https://github.com/OptimalBits/bull/blob/master/REFERENCE.md#queue) the Queue constructor has two formats:

~~~~
Queue(queueName: string, redisPort: number, redisHost: string, redisOpts?: RedisOpts): Queue
Queue(queueName: string, redisConnectionString: string, redisOpts? RedisOpts): Queue
~~~~

But the source:
~~~~
// Queue(name: string, url?, opts?)
var Queue = function Queue(name, url, opts){...}
~~~~

So the first options now seems invalid. Instead the code requires:

~~~~
Queue(queueName: string, redisConnectionString: string, redisOpts? RedisOpts): Queue
Queue(queueName: string, redisOpts?: RedisOpts): Queue
~~~~

2). Again, but the docs, the "redisOpts" param implies it references the options passed directly to ioredis:

~~~~
redisOptions {Object} Options to pass to the redis client. https://github.com/luin/ioredis/blob/master/API.md#new-redisport-host-options
~~~~

But the code requires a specific undocumented field "redis" that is passed to ioredis.

~~~~
new Queue({ redis: {...} });
~~~~

3). If I set the redis `retryStrategy` option, the method is called twice per retry.

  static DEF_OPTS = {
    host: '127.0.0.1',
    port: 6379,

    redis: {
      db: 0,
      retryStrategy: (attempt) => {
        logger.warn('Redis connection retry:', attempt);
        return Math.min(attempt * 100, 3000);
      }
    }
  };

I think this is probably because the "worker" has two connections? But per https://github.com/OptimalBits/bull/issues/526 I think there should be a way to configure a client that only has a connection for submitting events.

4). on('ready") never seems to get called (and in the code `emit('ready')` isn't called anywhere.

**Comment by manast** [comment]
1 and 2 and 4:
I think you are checking the documentation for 2.2.6 but the sourcecode of 3.x, since the constructor matches the documentation in master: https://github.com/OptimalBits/bull/blob/master/lib/queue.js#L66

Before 3.0 is released the documentation will be updated.

3) Which version of bull are you testing?

**Comment by richburdon** [comment]
Sorry, I didn't realize 3 was still in alpha, but now see your README
clearly references that.

**Comment by manast** [comment]
We have now merged the 3.0 documentation to master so we can close this issue.
